Refactors facts about Java version

petems opened the pull request:
These facts have since been refactored, ported these changes from https://github.com/puppetlabs/puppetlabs-java/

petems committed:
Take upstream java fixes and put them here

Got some advice from some Puppetlabs people on how to do facts better! :)

petems committed:
Refactors facts about Java version

These facts have since been refactored, ported these changes from https://github.com/puppetlabs/puppetlabs-java/

mikemcquaid commented on lib/facter/java_major_version.rb:
```diff
@@ -0,0 +1,20 @@
+    java_version = Facter.value(:java_version)
+    java_patch_level = java_version.strip.split('_')[0].split('.')[1] unless java_version.nil?
+  end
+end
```
Needs a newline.

petems replied:
Fixed

mikemcquaid commented:
Hmm, I'm wondering if/why we need this for Java 8 support when we can just do version comparisons on the existing fact?

petems committed:
Improve fact to use java_home libexec

Catches edge-case where which java is resolved, but Java is not actually installed!

petems commented:
@mikemcquaid We don't _need_ it, as you can perform regex's on the current fact, it just makes it a little easier to have Facter/Ruby code to perform that string manipulation :smile:

mikemcquaid commented:
@petems Could I see some sample code that you'd like to use this? Thanks!

petems commented:
Been a while, gonna close this
